src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/handlers/ReconfigurationCaseEventHandler.java

```diff
@@ -0,0 +1,128 @@
+package uk.gov.hmcts.reform.wacaseeventhandler.handlers;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.wacaseeventhandler.clients.TaskManagementApiClient;
+import uk.gov.hmcts.reform.wacaseeventhandler.clients.WorkflowApiClient;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.camunda.CancellationActions;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.camunda.DmnValue;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.camunda.request.EvaluateDmnRequest;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.camunda.response.CancellationEvaluateResponse;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.camunda.response.EvaluateDmnResponse;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.camunda.response.EvaluateResponse;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.ccd.message.EventInformation;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.model.MarkTaskToReconfigureTaskFilter;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.model.TaskFilter;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.model.TaskFilterOperator;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.model.TaskOperation;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.model.TaskOperationName;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.model.TaskOperationRequest;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static uk.gov.hmcts.reform.wacaseeventhandler.domain.camunda.DmnAndMessageNames.TASK_CANCELLATION;
+import static uk.gov.hmcts.reform.wacaseeventhandler.domain.camunda.DmnValue.dmnStringValue;
+
+@Slf4j
+@Service
+@Order(3)
+@SuppressWarnings({"PMD.DataflowAnomalyAnalysis", "unchecked"})
+public class ReconfigurationCaseEventHandler implements CaseEventHandler {
+
+    private final AuthTokenGenerator serviceAuthGenerator;
+    private final WorkflowApiClient workflowApiClient;
+    private final TaskManagementApiClient taskManagementApiClient;
+
+    public ReconfigurationCaseEventHandler(AuthTokenGenerator serviceAuthGenerator, WorkflowApiClient workflowApiClient,
+                                           TaskManagementApiClient taskManagementApiClient) {
+        this.serviceAuthGenerator = serviceAuthGenerator;
+        this.workflowApiClient = workflowApiClient;
+        this.taskManagementApiClient = taskManagementApiClient;
+    }
+
+    @Override
+    public List<? extends EvaluateResponse> evaluateDmn(EventInformation eventInformation) {
```

Comment:
This class can extend from CancellationCaseEventHandler as you don't need to duplicate evaluateDmn method then

Reply:
Why do you think so? Cancellation is different event, Reconfiguration is different event, no way related its just like other events Warning & Initalization